### PR TITLE
cli (ticdc): [create,query, update]changefeed cli use api

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #xxx

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: pkg/cmd/factory/factory.go [hunk 1]
```diff
@@ -107,25 +106,25 @@ func (c *ClientFlags) AddFlags(cmd *cobra.Command) {
...
 }
 
-// Validate makes sure provided values for ClientFlags are valid.
-func (c *ClientFlags) Validate() error {
-	tlsConfig, err := c.ToTLSConfig()
-	if err != nil {
-		return errors.Annotate(err, "fail to validate TLS settings")
-	}
+// // Validate makes sure provided values for ClientFlags are valid.
+// func (c *ClientFlags) Validate() error {
```
Comment (sdojjy) [inline pkg/cmd/factory/factory.go:116]: remove these lines

File: cdc/api/v2/changefeed.go [hunk 2]
```diff
@@ -37,14 +40,14 @@ func (h *OpenAPIV2) CreateChangefeed(c *gin.Context) {
 
 	ctx := c.Request.Context()
 
-	config := &ChangefeedConfig{ReplicaConfig: getDefaultReplicaConfig()}
+	config := &ChangefeedConfig{ReplicaConfig: GetDefaultReplicaConfig()}
```
Comment (sdojjy) [inline cdc/api/v2/changefeed.go:43]: if pd endpoints is empty, use the default upstream

File: cdc/api/v2/changefeed.go [hunk 3]
```diff
@@ -94,21 +97,109 @@ func (h *OpenAPIV2) VerifyTable(c *gin.Context) {
...
+	if newCfInfo != nil {
+		err = h.capture.EtcdClient.SaveChangeFeedInfo(ctx, newCfInfo, changefeedID)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+	}
+
+	if newUpInfo != nil {
+		err = h.capture.EtcdClient.SaveChangeFeedInfo(ctx, newCfInfo, changefeedID)
```
Comment (sdojjy) [inline cdc/api/v2/changefeed.go:174]: should be SaveUpstreamInfo?

Comment (asddongmen) [reply cdc/api/v2/changefeed.go]: yes

Comment (asddongmen) [reply cdc/api/v2/changefeed.go]: fixed

**Pull request merged**